lib/cache/inventory/inventory_cache.go

```diff
@@ -143,6 +153,47 @@ func (u *inventoryInstance) getTypeKey() bytestring {
 	return bytestring(ordered.Encode(u.getKind(), name, id))
 }
 
+// getVersionKey returns the composite key for sorting by version using semver ordering.
+func (u *inventoryInstance) getVersionKey() bytestring {
+	var versionStr, name, id string
+	if u.isInstance() {
+		versionStr = u.instance.Spec.Version
+		name = u.instance.GetHostname()
+		id = u.instance.GetName()
+	} else {
+		if u.bot.Status != nil && len(u.bot.Status.LatestHeartbeats) > 0 {
+			versionStr = u.bot.Status.LatestHeartbeats[0].Version
+		}
+		name = u.bot.GetSpec().GetBotName()
+		id = u.bot.GetSpec().GetInstanceId()
+	}
+
+	// If version is empty, treat it as 0.0.0
+	if versionStr == "" {
+		return bytestring(ordered.Encode(uint64(0), uint64(0), uint64(0), uint64(1), "", name, id))
+	}
+
+	// Strip "v" prefix if it's there, eg. v1.2.3 -> 1.2.3
+	versionStr = strings.TrimPrefix(versionStr, "v")
+
+	// Parse as semver
+	v, err := semver.NewVersion(versionStr)
+	if err != nil {
+		// Invalid semvers sort after all other versions
+		return bytestring(ordered.Encode(uint64(1<<63), versionStr, name, id))
+	}
+
+	// Ensure prereleases sort before releases
+	isRelease := uint64(1)
+	prerelease := ""
+	if v.PreRelease != "" {
+		isRelease = 0
+		prerelease = string(v.PreRelease)
+	}
```

Comment:
You could encode blank prereleases as `ordered.Inf` so they're greater than any string.